src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -35,16 +35,16 @@ public class Transaction : IEquatable<Transaction>, IInventory, IInteroperable
         private byte[] script;
         private Witness[] witnesses;
 
-        private static readonly long SignatureContractCost =
-            ApplicationEngine.OpCodePrices[OpCode.PUSHDATA1] * 2 +
-            ApplicationEngine.OpCodePrices[OpCode.PUSHNULL] +
-            ApplicationEngine.OpCodePrices[OpCode.SYSCALL] +
+        private static long SignatureContractCost(uint baseExecFee) =>
+            baseExecFee * (ApplicationEngine.OpCodePrices[OpCode.PUSHDATA1] * 2 +
+         ApplicationEngine.OpCodePrices[OpCode.PUSHNULL] +
+         ApplicationEngine.OpCodePrices[OpCode.SYSCALL]) +
             ApplicationEngine.ECDsaVerifyPrice;
```

Comment:
`ECDsaVerifyPrice` doesn't need to be multiplied by `baseExecFee`?

Reply:
This one specifically concentrates on opcodes. Interops are a bit more complex, some (like ECDsaVerifyPrice) are clearly CPU-bound, some (like GetBlock) are more about storage access/size. Anyway, we have another PR for interops and this can be handled there if needed.

Reply:
It'll be added in another PR https://github.com/neo-project/neo/pull/2063

Reply:
Maybe SYSCALLs should have two prices, one for calculation and the other for storage.

Reply:
Yeah, I think it's good to have this distinction but let's do that in another PR because it's a separate (even though related) problem.

Reply:
OK, I've implemented a draft of that just to see how it could look like (not changing specific interop prices), tests are not updated for now.